config/kernel.m4

```diff
@@ -461,15 +461,13 @@ AC_DEFUN([ZFS_AC_KERNEL_TEST_MODULE], [
 	ZFS_LINUX_TRY_COMPILE([], [], [
 		AC_MSG_RESULT([yes])
 	],[
-		AC_MSG_RESULT([no])
 		if test "x$enable_linux_builtin" != xyes; then
+			AC_MSG_RESULT([no])
 			AC_MSG_ERROR([
 	*** Unable to build an empty module.
 			])
 		else
-			AC_MSG_ERROR([
-	*** Unable to build an empty module.
-	*** Please run 'make scripts' inside the kernel source tree.])
+			AC_MSG_RESULT([unneeded, using --enable-linux-builtin])
```

Comment:
Setting `KBUILD_MODPOST_NOFINAL='yes'` for this test build should also avoid the issue.  This is effectively what the previous version did in `ZFS_LINUX_COMPILE_IFELSE` by setting `modpost=true` so it always passed the modpost phase when `--enable-linux-builtin` was set.

I believe something like this should work when `--enable_linux_builtin`.  Then we could preserve the existing, usually helpful, error message on failure.  Untested patch:

```diff
diff --git a/config/kernel.m4 b/config/kernel.m4
index ec8f6a5..4729c1e 100644
--- a/config/kernel.m4
+++ b/config/kernel.m4
@@ -759,7 +759,8 @@ AC_DEFUN([ZFS_LINUX_COMPILE_IFELSE], [
 
        ZFS_LINUX_CONFTEST_MAKEFILE([conftest], [no],
            [m4_ifvaln([$5], [-I$PWD/build/conftest], [])])
-       ZFS_LINUX_COMPILE([build/conftest], [$2], [$3], [$4], [], [])
+       ZFS_LINUX_COMPILE([build/conftest], [$2], [$3], [$4],
+           [$enable-linux-builtin], [])
 ])
 
 dnl #
```